### DISCUSSION TITLE
Merging of transitions styles

### DISCUSSION THREAD

**Discussion by flugg** [emotion-js/emotion#2330] [Q&A] [answered]
If you have a `Foo` component that sets transitions, e.g:

```typescript
export const Foo = styled.div(() => ({
  transition: 'background 0.2s, border 0.2s',
  // ...
}));
```

And a `Bar` component that encapsulates `Foo`. How can `Bar` define additional CSS for `Foo` that adds a new transition without overriding all the already set transitions on `Foo`? I want to do something along these lines:

```typescript
export const Bar = () => (
  <Foo css={{
    transition: 'box-shadow 0.2s',
    // ...
  }} />
);
```

And end up with a merged transition on `Foo`:

```css
transition: 'background 0.2s, border 0.2s, box-shadow 0.2s',
```

Is this even possible? If not, any other ideas on how to achieve something similar?

Thanks!

**Comment by Andarist** [comment, marked as answer]
Its not possible - i would pass the required info outside of css prop and only dump the whole thing to css at the very bottom
